js/data/store_helper.js

```diff
@@ -37,6 +37,10 @@ function queryByOptions(query, options, isCountQuery) {
 
     const filter = options.filter;
 
+    if(options?.compareOptions) {
+        query.setCompareOptions?.(options?.compareOptions);
```

Comment:
We don't need second ```?``` here:
```suggestion
        query.setCompareOptions?.(options.compareOptions);
```